pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -513,8 +513,25 @@ func configFromProviderSpec(providerSpec machinev1.ProviderSpec) (*bmv1alpha1.Ba
 	return &config, nil
 }
 
-// updateMachineStatus updates a machine object's status.
-func (a *Actuator) updateMachineStatus(ctx context.Context, machine *machinev1.Machine, host *bmh.BareMetalHost) error {
+// updateMachine updates a machine object's status.
+func (a *Actuator) updateMachine(ctx context.Context, machine *machinev1.Machine, host *bmh.BareMetalHost) error {
+	machineLabels := machine.ObjectMeta.GetLabels()
+	if machineLabels == nil {
+		machineLabels = make(map[string]string)
+	}
+	machineLabels["machine.openshift.io/instance-type"] = host.Status.HardwareProfile
+	machine.ObjectMeta.SetLabels(machineLabels)
+	machineAnnotations := machine.ObjectMeta.GetAnnotations()
+	if machineAnnotations == nil {
+		machineAnnotations = make(map[string]string)
+	}
+	machineAnnotations["machine.openshift.io/instance-state"] = strings.Replace(string(host.Status.Provisioning.State), " ", "_", -1)
+	machine.ObjectMeta.SetAnnotations(machineAnnotations)
+	err := a.client.Update(ctx, machine)
```

Comment:
After this point the machine will be stale, so the attempt to store the status on line 540 is going to result in an error. It would be better to anticipate that by separating these 2 update steps into separate functions that report whether they did any work. That way the caller in the reconcile loop knows that it should return and reconcile again.